### DISCUSSION TITLE
How can I create a decorator which adds type-hinted arguments to path functions?

### DISCUSSION THREAD

**Discussion by tkeith** [fastapi/fastapi#9063] [Questions]
### First check

* [ X ] I used the GitHub search to find a similar issue and didn't find it.
* [ X ] I searched the FastAPI documentation, with the integrated search.
* [ X ] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

I am trying to create a decorator which generates FastAPI paths, while still maintaining the ability for the decorated function's signature to dictate input data from FastAPI (query parameters, URL parts, etc). Here is a sample of what's not working for me -- a decorator with the goal of taking arbitrary view functions, and generating two GET handlers each: one that generates HTML (failing because it's unable to accept the fastapi.Request object, as the signature is overridden by f's signature), and one that generates TXT with simple key authentication (this one fails similarly due to the key parameter).

```
def page(path):
    def decr(f):
        @app.get(path + '.html')
        @functools.wraps(f)
        def html(request: fastapi.Request, *args, **kwargs):
            text = f(*args, **kwargs)
            return templates.TemplateResponse("template.html", {"request": request, "body": text})

        @app.get(path + '.txt')
        @functools.wraps(f)
        def txt(key: str, *args, **kwargs):
            if key != 'super-secret':
                return 'Incorrect key'
            else:
                return f(*args, **kwargs)
    return decr

@page('/foo')
def foo(name: str):
    return f'Hello {name}!'
```

I feel like I might be missing something simple here, as this is something I used to do in Flask, but I'm new to Python type-hinting so I don't know all the ins-and-outs of how type hints interact with decorators (yet). Any help is appreciated :)

**Comment by sschiessl-bcp** [comment]
Following
